tc/core/tc2halide.cc

```diff
@@ -852,6 +854,17 @@ HalideComponents translateDef(const lang::Def& def, bool throwWarnings) {
         body =
             substitute(rest->name, Variable::make(Int(32), first->name), body);
         body = mutate(Block::make(first->body, body));
+        if (restAsBlock) {
+          return Block::make(
+              For::make(
+                  first->name,
+                  first->min,
+                  first->extent,
+                  first->for_type,
+                  first->device_api,
+                  body),
```

Comment:
This looks like code duplication.
Can't you create the For first and then figure out if you need to put a block
around it?